src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/ApplicationParts/ApplicationAssembliesProvider.cs

```diff
@@ -147,135 +147,27 @@ private AssemblyItem GetAssemblyItem(Assembly assembly)
-                foreach (var library in compileLibraries)
-                {
-                    if (compileLibrariesWithNoDuplicates.ContainsKey(library.Name))
-                    {
-                        throw new InvalidOperationException(Resources.FormatCandidateResolver_DifferentCasedReference(library.Name));
-                    }
-
-                    compileLibrariesWithNoDuplicates.Add(library.Name, CreateDependency(library, referenceAssemblies));
-                }
-
-                _compileLibraries = compileLibrariesWithNoDuplicates;
-            }
-
-            public bool IsCandidate(Library library)
-            {
-                var classification = ComputeClassification(library.Name);
-                return classification == DependencyClassification.ReferencesMvc;
-            }
-
-            private Dependency CreateDependency(Library library, ISet<string> referenceAssemblies)
-            {
-                var classification = DependencyClassification.Unknown;
-                if (referenceAssemblies.Contains(library.Name))
-                {
-                    classification = DependencyClassification.MvcReference;
-                }
-
-                return new Dependency(library, classification);
-            }
-
-            private DependencyClassification ComputeClassification(string dependency)
-            {
-                if (!_compileLibraries.ContainsKey(dependency))
-                {
-                    // Library does not have runtime dependency. Since we can't infer
-                    // anything about it's references, we'll assume it does not have a reference to Mvc.
-                    return DependencyClassification.DoesNotReferenceMvc;
-                }
-
-                var candidateEntry = _compileLibraries[dependency];
-                if (candidateEntry.Classification != DependencyClassification.Unknown)
-                {
-                    return candidateEntry.Classification;
-                }
-                else
-                {
-                    var classification = DependencyClassification.DoesNotReferenceMvc;
-                    foreach (var candidateDependency in candidateEntry.Library.Dependencies)
-                    {
-                        var dependencyClassification = ComputeClassification(candidateDependency.Name);
-                        if (dependencyClassification == DependencyClassification.ReferencesMvc ||
-                            dependencyClassification == DependencyClassification.MvcReference)
-                        {
-                            classification = DependencyClassification.ReferencesMvc;
-                            break;
-                        }
-                    }
-
-                    candidateEntry.Classification = classification;
-
-                    return classification;
+                    throw new InvalidOperationException(Resources.FormatCandidateResolver_DifferentCasedReference(library.Name));
```

Comment:
I assume this can still happen if you have multiple framework references in a single project so let this be.